### PR TITLE
Add @Nullable and @NonNull Annotations

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - This PR adds nullability annotations to provide better Kotlin support.
 - Removed `Context` parameter from `CardClient#tokenize()`

 ### Checklist

 - [x] Added a changelog entry

- @sarahkoop
- @sshropshire

### CODE REVIEW

File: Card/src/main/java/com/braintreepayments/api/AuthenticationInsight.java [hunk 1]
```diff
@@ -13,33 +16,30 @@
...
-            regulationEnv = regulationEnv.toLowerCase();
-        }
-
-        authInsight.regulationEnvironment = regulationEnv;
+        return new AuthenticationInsight(regulationEnv);
+    }
 
-        return authInsight;
+    AuthenticationInsight(String regulationEnvironment) {
+        this.regulationEnvironment = regulationEnvironment;
```
Comment (sshropshire) [inline Card/src/main/java/com/braintreepayments/api/AuthenticationInsight.java:42]: Moving `regulationEnvironment` property to the constructor allows us to make the property `@NonNull` on AuthenticationInsight.

File: BraintreeCore/src/main/java/com/braintreepayments/api/PostalAddress.java [hunk 2]
```diff
@@ -4,6 +4,9 @@
 import android.os.Parcelable;
 import android.text.TextUtils;
 
+import androidx.annotation.NonNull;
```
Comment (jplukarski) [inline BraintreeCore/src/main/java/com/braintreepayments/api/PostalAddress.java:7]: It looks like nothing is set as `@NonNull` in this file. Does this still need to be imported here?

File: BraintreeCore/src/main/java/com/braintreepayments/api/PreferredPaymentMethodsCallback.java [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 import android.content.Context;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
```
Comment (jplukarski) [inline BraintreeCore/src/main/java/com/braintreepayments/api/PreferredPaymentMethodsCallback.java:6]: Same question here, but this time it looks like only `@NonNull` is used.

**Review by demerino — APPROVED** [review]
Nice change update!

File: PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancing.java [hunk 4]
```diff
@@ -3,6 +3,9 @@
 import android.os.Parcel;
 import android.os.Parcelable;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
```
Comment (jplukarski) [inline PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancing.java:7]: Same questions as before, I don't see `@Nullable` being used in this file.

File: Venmo/src/main/java/com/braintreepayments/api/VenmoAccountNonce.java [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 import android.os.Parcel;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
```
Comment (jplukarski) [inline Venmo/src/main/java/com/braintreepayments/api/VenmoAccountNonce.java:6]: Same as previous questions for `@Nullable`

File: Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java [hunk 6]
```diff
@@ -1,5 +1,8 @@
 package com.braintreepayments.api;
 
+import androidx.annotation.NonNull;
```
Comment (jplukarski) [inline Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java:3]: Same as my other questions for `@NonNull` here

File: VisaCheckout/src/main/java/com/braintreepayments/api/VisaCheckoutNonce.java [hunk 7]
```diff
@@ -2,6 +2,9 @@
 
 import android.os.Parcel;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
```
Comment (jplukarski) [inline VisaCheckout/src/main/java/com/braintreepayments/api/VisaCheckoutNonce.java:6]: Same question here with `@Nullable`

**Review by jplukarski — APPROVED** [review]
Left a few comments but they all are related and could just be my admittedly limited understanding of Android and java import statements 🙃

**Comment by sarahkoop** [comment]
> Left a few comments but they all are related and could just be my admittedly limited understanding of Android and java import statements 🙃

Good catch - these were all unused imports! Cleaned them up and fixed one other typo I found while testing 👍

**Pull request merged**